desc/objectives/_geometry.py

```diff
@@ -459,6 +459,290 @@ def compute(self, params, constants=None):
+        verbose : int, optional
+            Level of output.
+
+        """
+        # local import to avoid circular import
+        from desc.coils import CoilSet, MixedCoilSet
+
+        is_mixed_coils = False
+        if isinstance(self.things[0], CoilSet):
+            if isinstance(self.things[0], MixedCoilSet):
+                is_mixed_coils = True
+                coil = self.things[0].coils
+                self._dim_f = len(coil)
+            else:
+                coil = self.things[0].coils[0]
+                self._dim_f = 1
+        else:
+            coil = self.things[0]
+            self._dim_f = 1
+
+        if self._grid is None:
+            # TODO: raise error if grid, transforms are not the same size as mixed coils
+            self._grid = (
+                LinearGrid(N=32)
+                if not is_mixed_coils
+                else [LinearGrid(N=32)] * self._dim_f
+            )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        if is_mixed_coils:
+            transforms = [
+                get_transforms(self._data_keys, obj=coil[i], grid=self._grid[i])
+                for i in range(self._dim_f)
+            ]
+        else:
+            transforms = get_transforms(self._data_keys, obj=coil, grid=self._grid)
+
+        self._constants = {
+            "transforms": transforms,
+            "is_mixed_coils": is_mixed_coils,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute data of coil for given data key.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of the coil's degrees of freedom.
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self._constants.
+
+        Returns
+        -------
+        f : float or array of floats
+            Coil length.
+        """
+        if constants is None:
+            constants = self._constants
+
+        if constants["is_mixed_coils"]:
+            coils = self.things[0].coils
+            data = [
+                coils[i].compute(
+                    self._data_keys,
+                    params=params[i],
+                    transforms=constants["transforms"][i],
+                    grid=self._grid[i],
+                )
+                for i, coil in enumerate(coils)
+            ]
+        else:
+            coils = self.things[0]
+            data = coils.compute(
+                self._data_keys,
+                params=params,
+                transforms=constants["transforms"],
+                grid=self._grid,
+            )
+
+        return data
+
+
+class CoilLength(_CoilObjective):
+    """Coil length.
+
+    Parameters
```

Comment:
should document all args here since this is the docstring that the users will see